cache/key_test.go

```diff
@@ -12,47 +15,28 @@ func TestWithVersionSuffix(t *testing.T) {
 		version string
 	}
 	tests := []struct {
-		name string
-		args args
-		want string
+		name    string
+		args    args
+		want    string
+		wantErr string
 	}{
 		{
 			name: "happy path",
 			args: args{
 				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
 				version: "111101112110013",
 			},
-			want: "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e/111101112110013",
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			got := CalcKey(tt.args.key, tt.args.version)
-			assert.Equal(t, tt.want, got)
-		})
-	}
-}
-
-func TestTrimVersionSuffix(t *testing.T) {
-	type args struct {
-		versioned string
-	}
-	tests := []struct {
-		name string
-		args args
-		want string
-	}{
-		{
-			name: "happy path",
-			args: args{
-				versioned: "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e/111101112110013",
-			},
-			want: "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+			want: "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := TrimVersionSuffix(tt.args.versioned)
+			got, err := CalcKey(tt.args.key, tt.args.version, &config.ScannerOption{})
```

Comment:
Could you add cases where ScannerOption is not empty? I want to make sure that the key will be different if we pass `FilePatterns` and `PolicyPaths`. Also, I'd like to check if the key is the same when we pass `[]string{}`, not nil.

Reply:
okay, I was thinking of adding cases to this but wanted to make sure if everything else was on the right track first.

Reply:
Looks the right way 👍 . Let me describe the background. IaC config files might include sensitive information such as passwords and API keys. If we store them in the cache, it might increase the attack surface for attackers. So, I changed it not to store file contents in the cache. Instead, only the results are stored in the cache now. But let's say we scan config files in container images. If a user updates Rego policies, the result should be updated. We must not use old caches. In other words, cache keys must be different after policies are updated. That's why I added `PolicyPaths`, `DataPaths`, and `FilePatterns` for calculating cache keys.